index.js

```diff
@@ -169,7 +169,91 @@ class Pawn extends Piece {
   get icon() {
     return this.isWhite() ? "♙" : "♟";
   }
+  onMove(move) {
+    this.hasMoved = true;
+  }
+}
+
+class Rook extends Piece {
+  constructor(game, board, color, row, file) {
+    super(game, board, color);
+    this.row = row;
+    this.file = file;
+    this.hasMoved = false;
+  }
 
+  name = "Rook";
+
+  get icon() {
+    return this.isWhite() ? "♖" : "♜";
+  }
+
+  horizontalMove(row, file, arr, direction) {
+    const horizontal = direction === "right" ? 1 : -1;
+
+    for (let i = 1; i < Board.LANE_SIZE; i++) {
+      const horizDirection = [row, file + i * horizontal];
+      if (
+        this.isValidSquare(...horizDirection) &&
+        !this.isSquareOccupied(...horizDirection)
+      ) {
+        arr.push(Move.fromSquare(horizDirection, this));
+      } else if (
+        this.isValidSquare(...horizDirection) &&
+        this.isSquareOccupied(...horizDirection)
+      ) {
+        if (this.getSquareContent(...horizDirection).color !== this.color) {
+          arr.push(
+            Move.fromSquare(
+              horizDirection,
+              this,
+              this.getSquareContent(...horizDirection)
+            )
+          );
+        }
+        return arr;
+      }
+    }
```

Comment:
Suggestion to fix a bug and clean up the control flow (i.e. if-else statements). 

As it stands the `return arr` statement will only be hit if an occupiedSquare condition is `true`. That means that if a square in this direction is not occupied the output of this function is `undefined` (because an explicit `return` statement won't be hit). On the other hand, it has the positive effect of ending the loop early when a piece is encountered, so we should move the `return` statement outside of the loop and replace it with a `break` statement (exits the loop immediately when hit). 

```suggestion

    for (let i = 1; i < Board.LANE_SIZE; i++) {
      const horizDirection = [row, file + i * horizontal];
      if (this.isValidSquare(...horizDirection)) {
        if (!this.isSquareOccupied(...horizDirection)) {
          arr.push(Move.fromSquare(horizDirection, this));
        } else {
          const otherPiece = this.getSquareContent(...horizDirection)
          if (otherPiece.color !== this.color) {
            arr.push(
              Move.fromSquare(
                horizDirection,
                this,
                otherPiece
              )
            );
          } else {
            break;
          }
        }
      }
    }
    return arr;

```